pkg/apis/kubebind/v1alpha1/apiservicebinding_types.go

```diff
@@ -87,8 +87,135 @@ type APIServiceBindingSpec struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="kubeconfigSecretRef is immutable"
 	KubeconfigSecretRef ClusterSecretKeyRef `json:"kubeconfigSecretRef"`
+
+	// permissionClaims records decisions about permission claims requested by the API service provider.
+	// Individual claims can be accepted or rejected. If accepted, the API service provider gets the
+	// requested access to the specified resources in this workspace. Access is granted per
+	// GroupResource, identity, and other properties.
+	//
+	// +optional
+	PermissionClaims []AcceptablePermissionClaim `json:"permissionClaims,omitempty"`
 }
 
+type AcceptablePermissionClaim struct {
+	PermissionClaim `json:",inline"`
+
+	// state indicates if the claim is accepted or rejected.
+
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=Accepted;Rejected
+	State AcceptablePermissionClaimState `json:"state"`
+}
+
+type AcceptablePermissionClaimState string
+
+const (
+	ClaimAccepted AcceptablePermissionClaimState = "Accepted"
+	ClaimRejected AcceptablePermissionClaimState = "Rejected"
+)
+
+// PermissionClaim identifies an object by GR and identity hash.
+// Its purpose is to determine the added permissions that a service provider may
+// request and that a consumer may accept and allow the service provider access to.
+//
+// +kubebuilder:validation:XValidation:rule="!(self.create.donate && self.adopt)",message="donate and adopt are mutually exclusive"
+type PermissionClaim struct {
+	GroupResource `json:","`
+
+	Version string `json:"version"`
+
+	// selector selects which resources are affected by this claim.
+	Selector ResourceSelector `json:"selector,omitempty"`
+
+	// required indicates whether the APIServiceBinding will work if this claim is not accepted.
+	Required bool `json:"required"`
+
+	// Global claims global resources for the given group/resource.
+	// This is mutually exclusive with resourceSelector.
+	// +optional
+	Global bool `json:"global,omitempty"`
+
+	// only for owner Provider
+	//
+	// create determines whether the kube-bind konnector will sync matching objects from the
+	// provider side down to the consumer cluster.
+	Create CreateOptions `json:"create"`
```

Comment:
this must be a pointer. There is no need that a service provider claims creation.